src/index.js

```diff
@@ -1,64 +1,83 @@
-    completed: false,
-    description: 'wake up at 6am',
-  },
-  {
-    index: 6,
-    completed: false,
-    description: 'take shower',
-  },
-  {
-    index: 3,
-    completed: false,
-    description: 'wash dishes',
-  },
-  {
-    index: 7,
-    completed: false,
-    description: 'start daily activities',
-  },
-];
+const todo = new Todo();
 
-data = data.sort((a, b) => a.index - b.index);
 const populateData = () => {
+  if (localStorage.getItem('todo')) {
+    todo.setTodos(JSON.parse(localStorage.getItem('todo')));
+  }
+  const data = todo.getTodos();
   const list = document.getElementById('list');
   list.innerHTML = '';
   data.forEach((singledata) => {
     const li = document.createElement('li');
     const licheckbox = document.createElement('input');
     const inputelem = document.createElement('input');
     const span = document.createElement('span');
-    span.setAttribute('class', 'dots');
+    const garbage = document.createElement('span');
+    garbage.innerHTML = '&#128465;&#65039;';
+    span.setAttribute('class', `${singledata.index} bullet${singledata.index} rmv dots`);
+    garbage.setAttribute('class', `${singledata.index} bullet${singledata.index} rmv garbage`);
     span.innerHTML = '&#8285;';
+    garbage.classList.add('show');
     licheckbox.setAttribute('type', 'checkbox');
-    licheckbox.setAttribute('class', 'checkbox');
-    inputelem.setAttribute('class', 'inputbox');
+    licheckbox.setAttribute('class', `${singledata.index} checkbox`);
+    licheckbox.setAttribute('id', `${singledata.index}`);
+    inputelem.setAttribute('class', `${singledata.index} bullet${singledata.index} inputfields inputbox`);
     inputelem.setAttribute('type', 'text');
     inputelem.setAttribute('value', singledata.description);
+    if (singledata.completed === true) {
+      inputelem.classList.toggle('completed');
+    }
     li.appendChild(licheckbox);
     li.appendChild(inputelem);
+    li.appendChild(garbage);
     li.appendChild(span);
     list.appendChild(li);
   });
 };
 
 populateData();
+
+document.addEventListener('click', (event) => {
+  if (event.target.classList.contains('checkbox')) {
+    todo.complete(Number(event.target.classList[0]));
+    localStorage.setItem('todo', JSON.stringify(todo.getTodos()));
+    populateData();
+  }
+
+  if (event.target.classList.contains('dots') || event.target.classList.contains('inputfields')) {
+    document.querySelector(`.${event.target.classList[1]}.garbage`).classList.remove('show');
+    document.querySelector(`.${event.target.classList[1]}.dots`).classList.add('show');
+  }
+
+  if (event.target.classList.contains('garbage')) {
+    todo.removeTodo(Number(event.target.classList[0]));
+    localStorage.setItem('todo', JSON.stringify(todo.getTodos()));
+    populateData();
+  }
+
+  if (event.target.classList.contains('clear-button')) {
+    todo.clearCompleted();
+    localStorage.setItem('todo', JSON.stringify(todo.getTodos()));
+    populateData();
+  }
+});
```

Comment:
- [ ] When removing a task you should update the indexes of all the remaining tasks as required please make sure to fix that 
![image](https://user-images.githubusercontent.com/61790587/172941869-888e3eb8-c920-4656-b849-83200b8b9c5f.png)
![image](https://user-images.githubusercontent.com/61790587/172941884-704116dd-5637-4696-bea7-a65d2b3672a6.png)